### ISSUE TITLE
Error when adding gin remote as common data source

### ISSUE THREAD

**Issue by mslw** [datalad/datalad#6195]
In trying to follow the https://handbook.datalad.org/en/latest/basics/101-139-gin.html#using-gin-as-a-data-source-behind-the-scenes I am unable to configure gin remote as a common data source.

```
datalad siblings add -d . --name gin --url git@gin.g-node.org:/msz/gin-and-git.git
[INFO   ] Could not enable annex remote gin. This is expected if gin is a pure Git remote, or happens if it is not accessible. 
[WARNING] Could not detect whether gin carries an annex. If gin is a pure Git remote, this is expected.  
.: gin(-) [git@gin.g-node.org:/msz/gin-and-git.git (git)]

mszczepanik@bnbnb64 in ~/Documents/tmp/gin-and-git on git:main
❱ datalad siblings                                                                                                                      1 !
.: here(+) [git]
[WARNING] Could not detect whether gin carries an annex. If gin is a pure Git remote, this is expected.  
.: gin(-) [git@gin.g-node.org:/msz/gin-and-git.git (git)]

mszczepanik@bnbnb64 in ~/Documents/tmp/gin-and-git on git:main
❱ datalad push --to gin
copy(ok): hubert-neufeld-j-udI4zim2E-unsplash.jpg (file) [to gin...]                                                                        
copy(ok): jay-ruzesky-9zTafGVsv-c-unsplash.jpg (file) [to gin...]                                                                           
copy(ok): paul-carroll-Y-nyDv3TWm0-unsplash.jpg (file) [to gin...]                                                                          
publish(ok): . (dataset) [refs/heads/git-annex->gin:refs/heads/git-annex 7571a4c..1be87df]                                                  
publish(ok): . (dataset) [refs/heads/main->gin:refs/heads/main [new branch]]                                                                
action summary:                                                                                                                             
  copy (ok: 3)
  publish (ok: 2)

mszczepanik@bnbnb64 in ~/Documents/tmp/gin-and-git on git:main
❱ datalad siblings     
.: here(+) [git]
.: gin(+) [git@gin.g-node.org:/msz/gin-and-git.git (git)]

mszczepanik@bnbnb64 in ~/Documents/tmp/gin-and-git on git:main
❱ datalad siblings add -d . --name gin-update --pushurl git@gin.g-node.org:/msz/gin-and-git.git --url https://gin.g-node.org/msz/gin-and-git --as-common-datasrc gin
CommandError: 'git -c diff.ignoreSubmodules=none annex initremote gin type=git location=https://gin.g-node.org/msz/gin-and-git autoenable=true -c annex.dotfiles=true' failed with exitcode 1 under /home/mszczepanik/Documents/tmp/gin-and-git
git-annex: There is already a remote named "gin"

mszczepanik@bnbnb64 in ~/Documents/tmp/gin-and-git on git:main
❱ datalad siblings                                                                                                                      1 !
.: here(+) [git]
.: gin(+) [git@gin.g-node.org:/msz/gin-and-git.git (git)]
.: gin-update(+) [https://gin.g-node.org/msz/gin-and-git (git)]
```

Debian GNU/Linux 11 (bullseye)
Datalad version 0.15.0
Git annex version 8.20210903-1~ndall+1

**Comment by mih** [comment]
You have to choose a different name. There cannot be two remotes of any type that have the same name.

This is what the error message says. Do you think it should be rephrased?

**Comment by mslw** [comment]
But the datalad siblings gets `--name gin-update` (new name) and then points to an existing one in `--as-common-datasrc gin`. Isnt' that how these arguments are supposed to be?

**Comment by mih** [comment]
But you created a `gin` remote already as the first step. It cannot have two

**Comment by adswa** [comment]
It works when I don't register ``gin`` as a sibling before hand, but just create it via webinterface.

```
(handbook2) adina@muninn in /tmp
❱ datalad create abcdef                                                                                         1 !
[INFO   ] Creating a new annex repo at /tmp/abcdef 
create(ok): /tmp/abcdef (dataset)
(handbook2) adina@muninn in /tmp
❱ cd abcdef                                                                                                     1 !
(handbook2) adina@muninn in /tmp/abcdef on git:master
❱ echo 234567 > file
(handbook2) adina@muninn in /tmp/abcdef on git:master
❱ datalad save
add(ok): file (file)                                                                                                
save(ok): . (dataset)                                                                                               
action summary:                                                                                                     
  add (ok: 1)
  save (ok: 1)
(handbook2) adina@muninn in /tmp/abcdef on git:master
❱ datalad siblings add \
 -d . \
 --name gin-update \
 --pushurl git@gin.g-node.org:/adswa/wargh.git \
 --url https://gin.g-node.org/adswa/wargh \
 --as-common-datasrc gin
[INFO   ] Could not enable annex remote gin-update. This is expected if gin-update is a pure Git remote, or happens if it is not accessible. 
[WARNING] Could not detect whether gin-update carries an annex. If gin-update is a pure Git remote, this is expected. Remote was marked by annex as annex-ignore. Edit .git/config to reset if you think that was done by mistake due to absent connection etc 
.: gin-update(-) [https://gin.g-node.org/adswa/wargh (git)]

```

I should probably make this more clear in the handbook. I first did the same thing @mslw did because I was using ``create-sibling-gin``

**Comment by mslw** [comment]
First time when I tried it without preexisting gin remote it didn't work, but now it does, as you expect. I'll think of it, maybe I made some mistake at the first try.

**Comment by mslw** [comment]
Thank you for clarification on the role of `--as-common-datasrc`. The command works as expected.

The error message is clear enough and I don't think it needs changing. The handbook also explains everything all right.

The only reason I reported a problem with an existing remote in place is that - I could swear - the first time I tried doing it a correct way I got a git annex error "could not find existing git remote with specified location". But now I cannot even replicate that one so I don't know what happened.

So it appears that my confusion was self-generated, even though I can't pinpoint how. Sorry to cause the alarm.
